.github/workflows/nightly_build.yml

```diff
@@ -40,53 +41,125 @@ jobs:
           poetry env use ${{ env.PYTHON_VERSION }}
           poetry install
 
-      - name: Create tag
-        id: create_tag
+      - name: Generate main nightly tag
+        id: generate_main_tag
         run: |
+          # NOTE: This outputs the tag with the `v` prefix.
+          MAIN_TAG="$(poetry run python ./scripts/ci/pypi_nightly_tag.py main)"
+          echo "main_tag=$MAIN_TAG" >> $GITHUB_OUTPUT
+          echo "main_tag=$MAIN_TAG"
+
+      - name: Check if main tag already exists
```

Comment:
if the main tag already existed due to a partial workflow, we need to pull that and use the base version associated with it